sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
```

Comment:
`-1`? So, if length is 0, the end is before offset? If length is 1, the range is empty? If length is 2, the range contains one element?

Regardless of `-1`, can range be negative? Can offset be negative? Can Offset point out of the bounds of the enclosing range? What if length ends up end of the range pointing outside of the enclosing range?

Reply:
The way .NET solved those questions is by not allowing negative length/offset, and only allowing length to be > 0, if it iis provided.
https://github.com/Azure/azure-sdk-for-net/blob/8b100f13b70aa00fdb69304500ea7e2a6a142322/sdk/core/Azure.Core/src/HttpRange.cs#L38-L41